drivers/gpio/Kconfig.intel

```diff
@@ -3,15 +3,24 @@
 # Copyright (c) 2018 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-menuconfig GPIO_INTEL_APL
-	bool "Intel Apollo Lake SoC GPIO"
+menuconfig GPIO_INTEL
+	bool "Intel Soc GPIO"
+	help
+	  Enable driver for Intel SoC GPIO
+
+config GPIO_INTEL_APL
```

Comment:
Why do you need the separate `GPIO_INTEL_APL` and `GPIO_INTEL_EHL` Kconfig options once you have the unified `GPIO_INTEL` option? It seems like you could drop the platform-specific options.

Reply:
These configs are being used currently to determine what value to initialize DT_DRV_COMPAT with.

Reply:
Right, and what I'm saying is that you should only have a single DT binding and a single compatible string value.

Reply:
Btw, it seems like there are no users of the `intel,apollo-lake-gpio.yaml` binding (at least I couldn't find any). So that should probably be removed and then the `intel,apl-gpio.yaml` binding should be renamed to `intel,gpio.yaml` to match the new unified driver.